### PR TITLE
Update golang.org/x/exp

### PR DESCRIPTION
**Pull request by muff1nman**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Updates golang.org/x/exp. This package has seen some breaking changes as of late. Updating to latest alleviates dependency conflicts.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/log/wrapper.go [hunk 1]
```diff
@@ -46,37 +47,32 @@ type Logger struct {
...
 func (l *Logger) Warn(msg string, args ...any) {
-	l.LogDepth(0, WarnLevel, msg, args...)
+	l.log.Log(context.TODO(), WarnLevel, msg, args...)
 }
 
 // Error logs an error message.
 func (l *Logger) Error(msg string, err error, args ...any) {
 	if err != nil {
-		args = append(args[:len(args):len(args)], slog.Any(slog.ErrorKey, err))
+		args = append(args[:len(args):len(args)], slog.Any("err", err))
```
Comment (wzshiming) [inline pkg/log/wrapper.go:73]: It seems that the logic of the `slog` has big changed

File: pkg/log/wrapper.go [hunk 2]
```diff
@@ -46,37 +47,32 @@ type Logger struct {
 	level Level // Level specifies a level of verbosity for V logs.
 }
 
-// LogDepth logs a message with the given level and depth.
-func (l *Logger) LogDepth(calldepth int, level Level, msg string, args ...any) {
```
Comment (wzshiming) [inline pkg/log/wrapper.go:50]: What does this `calldepth` now use instead?

`source` is not the correct log location.

``` console
{"time":"2023-05-04T14:01:53.85924416Z","level":"DEBUG","source":"/home/runner/work/kwok/kwok/pkg/log/wrapper.go:57","msg":"Load config","path":[],"err":"empty config"}
{"time":"2023-05-04T14:01:53.859318761Z","level":"DEBUG","source":"/home/runner/work/kwok/kwok/pkg/log/wrapper.go:57","msg":"No configuration","kind":"KwokctlConfiguration"}
{"time":"2023-05-04T14:01:53.859670064Z","level":"INFO","source":"/home/runner/work/kwok/kwok/pkg/log/wrapper.go:62","msg":"Cluster is creating","cluster":"cluster-binary-1-27-1"}
```

Comment (muff1nman) [reply pkg/log/wrapper.go:50]: I believe I have this fixed now, its unfortunate but had to replicate some of the slog.Logger logic into our own wrapper.

File: pkg/log/wrapper.go (hunk 1 above)
Comment (muff1nman) [reply pkg/log/wrapper.go:75]: Yeah it certainly has a decent amount of changes. Hopefully when it gets added to stdlib things will slow down :)

**Comment by muff1nman** [comment]
I can squash these commits into one, I just figured I'd keep them separate to contrast the two approaches.

**Comment by wzshiming** [comment]
Thank you for your contribution!

/approve
/lgtm
/label tide/merge-method-squash

**Pull request merged**